beeline/src/java/org/apache/hive/beeline/schematool/HiveSchemaTool.java

```diff
@@ -112,6 +121,40 @@ protected void execSql(String sqlScriptFile) throws IOException {
     }
   }
 
+  void replaceLocationForProtoLogTables(String sqlScriptFile) throws IOException {
+    TezConfiguration tezConf = new TezConfiguration(true);
+    String hiveProtoBaseDir = HiveConf.getVar(conf, HiveConf.ConfVars.HIVE_PROTO_EVENTS_BASE_PATH);
+    String tezProtoBaseDir = tezConf.get(TezConfiguration.TEZ_HISTORY_LOGGING_PROTO_BASE_DIR);
+
+    if (isEmpty(hiveProtoBaseDir)) {
+      throw new IOException("Hive conf variable hive.hook.proto.base-directory is not set for creating protologging tables");
+    }
+    if (isEmpty(tezProtoBaseDir)) {
+      throw new IOException("Tez conf variable tez.history.logging.proto-base-dir is not set for creating protologging tables");
+    }
+
+    Map<String, String> replacements = new HashMap<>();
+    replacements.put("_REPLACE_WITH_QUERY_DATA_LOCATION_", "hdfs:///" + hiveProtoBaseDir);
+    replacements.put("_REPLACE_WITH_APP_DATA_LOCATION_", "hdfs:///" + tezProtoBaseDir + "/app_data");
+    replacements.put("_REPLACE_WITH_DAG_DATA_LOCATION_", "hdfs:///" + tezProtoBaseDir + "/dag_data");
+    replacements.put("_REPLACE_WITH_DAG_META_LOCATION_", "hdfs:///" + tezProtoBaseDir + "/dag_meta");
```

Comment:
I presume we may also avoid hardcoding it for TEZ_HISTORY_LOGGING_PROTO_BASE_DIR.

Reply:
Sorry for the late update as I was away due to personal issues. I have updated the files by working on your suggestions.